compiler/testData/debug/localVariables/tryFinally10.kt

```diff
@@ -0,0 +1,55 @@
+// FILE: test.kt
+
+inline fun f(block: () -> Unit) {
+    try {
+        val z = 32
+        for (j in 0 until 1) {
+            return
+        }
+    } finally {
+        block()
+    }
+}
+
+var x: String = ""
+
+fun compute(): String {
+    try {
+        val y = 42
+        for (i in 0 until 1) {
+            f {
+                return "NON_LOCAL_RETURN"
+            }
+        }
+    } finally {
+        val s2 = "OK"
+        x = s2  // TODO: Why is `s2` not visible here?
+    }
+    return "FAIL"
+}
+
+fun box() {
+    val result = compute()
+    val localX = x
+}
+
+// The local variables `y` and `i` are visible in the finally block with old backend.
+// IGNORE_BACKEND: JVM
+
+// LOCAL VARIABLES
+// test.kt:32 box:
+// test.kt:17 compute:
+// test.kt:18 compute:
+// test.kt:19 compute: y:int=42:int
+// test.kt:20 compute: y:int=42:int, i:int=0:int
+// test.kt:4 compute: y:int=42:int, i:int=0:int, $i$f$f:int=0:int
+// test.kt:5 compute: y:int=42:int, i:int=0:int, $i$f$f:int=0:int
```

Comment:
Not for this change, but for all of these inline functions we actually should remove the locals from the caller. They are not lexically visible in the inline function and therefore should not show up there when debugging. However, I would like to look at that separately.

Reply:
Hmm, looks like the debugger filters them out, so apparently they are not a user visible issue (and for the VM the local is there so all is well).

Reply:
Still might be better to do correct ranges so that the behavior does not depend on the debugger used. On the other hand if you use a debugger that does not understand the inlining locals you are probably going to have a bit of a bad time debugging this anyway. :-\

Reply:
@madsager I can't catch your intention. `$i$f$functionName`, `$i$a$functionName`  variables are generated specially for debugger to distinguish inlined function code from inline lambda one. If any of call site finally blocks inserted into inline function than yes ranges for such variable should be corrected

Reply:
Oh, I wasn't talking about the inlining locals. Those are removed for the finally block and that is working as intended.

I was talking about the fact that `y` and `i` loclas are covering the instructions for the inlined code for f here. So, the locals from the method into which the inline function is inlined are in the locals table when you are in the inline function. That is not right as they are not in scope. It seems like the debugger is capable of filtering them out even though they are in the locals table. When in `f` in the intellij debugger I do not see `y` and `i`. However, in jdb they are visible for all the lines of `f`.

Reply:
Maybe they are there so that when in the inline function you could have two stack frames. One for caller and one for inline function callee. In the caller you would show `y` and `i`? The source maps should allow you to show full stack traces, but it doesn't look like IntelliJ is currently doing that. However, to support it, these locals would have to be there. So maybe this is working as intended?

Reply:
@madsager It's a good question, @yanex could you answer?

Reply:
The current belief is that the debugger should show call-site variables in inline lambdas (where their evaluation makes sense) and shouldn't show them in inline functions (just like in a non-inline function, you don't see variables from a caller). The logic is quite straightforward there: if the debugger recognizes we're inside an inline function but not in an inlined lambda (`$i$f` appears later than `$i$a`, or `$i$a` is absent), we only show local variables with the most `$iv` at the end.

We don't show call-site variables for non-inlined lambdas. However, the Java debugger in IntelliJ does that for simple cases (for instance, when the lambda runs in the same thread), so we'll probably also implement similar behavior rather sooner than later.

Speaking about stack traces for inline functions, we don't show them now, but we've already started experimenting in this direction. Without a doubt, we'll need access to call-site variables from any place in an inline function. Yet, the current scheme is also not ideal in less trivial cases. For instance, here we lose access to the outer `a`:

```
fun foo() {
    val a = 1
    "foo".apply {
        print(a)
        val a = 2 // a = 1 is not available anymore
        print(a)
    }
}
```